Profiler/DTXProfiler/ReactNativeProfiling/DTXReactNativeSampler.m

```diff
@@ -153,6 +159,78 @@ static void installDTXSignpostHook(JSContext* ctx)
 	};
 }
 
+static void installJSTraceEventsHook(JSContext* ctx)
+{
+	ctx[@"nativeTraceBeginSection"] = ^ (int tag, NSString* _name , id args)
+	{
+		static dispatch_once_t onceToken;
+		dispatch_once(&onceToken, ^{
+			sectionsRegularEvents = [[NSMutableArray alloc] init];
+		});
+		
+		DTXEventIdentifier identifier = __DTXProfilerMarkEventIntervalBegin(NSDate.date,@"JSSection",_name,@"",NO,YES,nil);
+		if(identifier)
+		{
+			@synchronized(sectionsRegularEvents) {
+				[sectionsRegularEvents addObject: identifier];
+			}
+		}
+	};
+	
+	ctx[@"nativeTraceEndSection"] = ^ (int tag)
+	{
+		DTXEventIdentifier identifier = [sectionsRegularEvents lastObject];
+		if(identifier)
+		{
+			__DTXProfilerMarkEventIntervalEnd(NSDate.date, identifier, DTXEventStatusCompleted, nil);
+			[sectionsRegularEvents removeLastObject];
+		}
+	};
+	
+	ctx[@"nativeTraceBeginAsyncSection"] = ^(int tag, NSString* name, int cookie)
+	{
+		static dispatch_once_t onceToken;
+		dispatch_once(&onceToken, ^{
+			asyncSections = [[NSMutableDictionary alloc] init];
+		});
+		
+		DTXEventIdentifier identifier = __DTXProfilerMarkEventIntervalBegin(NSDate.date,@"JSAsyncSection",name,@"",NO,YES,nil);
+		if(identifier)
+		{
+			@synchronized (asyncSections) {
```

Comment:
Please don't use `@synchronized` as it is slow. Instead, use `__eventDispatchQueue` to asynchronously implement the body of all the functions. This will ensure the two dictionaries are used correctly.
To keep the start/end dates as accurate as possible, obtain the date before using `dispatch_async`.